File: id_REVC/AH/003_AH_RevCompDNA.pl
#!/usr/bin/perl

use strict;
use warnings;

my $string="AAAACCCGGT";

#use translate reg ex.
$string =~ tr/ACGTT/TGCA/;

$string = reverse $string;

print $string."\n";

#[ahedley@Y90-AHED-L-D2 id_DNA]$ ./003_AH_RevCompDNA.pl 
#ACCGGGTTTT

File: id_REVC/AH/003_AH_RevCompDNA.py
#!/usr/bin/python2

from string import maketrans   # Required to call maketrans function.

intab = "ACGT"
outtab = "TGCA"
trantab = maketrans(intab, outtab)

str = "AAAACCCGGT";
print str.translate(trantab)[::-1]

# str the input string
# translate(trantab) translate it using the tran table
# [::-1] reverse it


#[ahedley@Y90-AHED-L-D2 id_DNA]$ ./003_AH_RevCompDNA.py
#ACCGGGTTTT

